### PR TITLE
GROWTH-143 fix check to use required partition by column and check al…

### PR DESCRIPTION
**Pull request by kwindau**
…l rows to make sure client only 1x in table

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-4016)

### CODE REVIEW

**Review by fbertsch — COMMENTED** [review]
Does it make more sense to remove the partitions & the partition requirement? I don't really see the purpose of them for a grain of per-client

**Comment by kwindau** [comment]
> Does it make more sense to remove the partitions & the partition requirement? I don't really see the purpose of them for a grain of per-client

I was debating it but I think this table could get big over time since it's going to basically have 1 row per every client, I was leaning towards keeping it so that when people query they have to specify, but in the checks force them to scan the whole table to make sure no clients end up in there twice

File: sql/moz-fx-data-shared-prod/google_ads_derived/conversion_event_categorization_v1/checks.sql [hunk 1]
```diff
@@ -1,2 +1,2 @@
 #warn
-{{ is_unique(["client_id"]) }}
+{{ is_unique(["client_id"]), "first_seen_date <= current_date" }}
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/google_ads_derived/conversion_event_categorization_v1/checks.sql:2]:
```suggestion
{{ is_unique(["client_id"], "first_seen_date <= current_date") }}
```

Comment (kik-kik) [reply sql/moz-fx-data-shared-prod/google_ads_derived/conversion_event_categorization_v1/checks.sql:2]: If I recall correctly, this is the correct place to have the closing bracket.

**Pull request merged**